launcher/FileSystem.cpp

```diff
@@ -232,22 +193,30 @@ bool deletePath(QString path)
     return OK;
 }
 
+bool trash(QString path)
+{
+#if QT_VERSION < QT_VERSION_CHECK(5, 15, 0)
+    return deletePath(path);
+#else
+    return QFile::moveToTrash(path) || deletePath(path);
+#endif
+}
```

Comment:
I'm not sure I like the idea of automatically falling back to deletion if moving to the trash doesn't work or isn't supported. Usually when you delete files in your file browser and it can't be moved to the trash for whatever reason, the system warns you first that whatever you're deleting will be gone permanently.

This may cause confusion (what if someone who's used to the AppImage migrates to a different package that uses an older Qt version, and deletes an instance expecting it to go to the trash?), which is quite risky when it comes to data deletion.

I'd suggest putting up a warning box asking for confirmation if moving to the trash fails for whatever reason, making it clear that the deletion can't be undone.

Reply:
I don't think users on an OS without a system trash would *expect* in the first place to have instances moved to trash.

Since not having a system trash is the main reason for `QFile::moveToTrash` to fail.

Reply:
Although the current messaging does say "IRREVERSIBLE" (I think) even though trashing would change that behavior.

Reply:
Or we can just say "the instance may be deleted permanently if your system does not have a trash"

Reply:
> I don't think users on an OS without a system trash would _expect_ in the first place to have instances moved to trash.
> 
> Since not having a system trash is the main reason for `QFile::moveToTrash` to fail.

Maybe, but there are other fairly common reasons depending on the OS. For example, on Windows I believe there is a maximum file size for items in the trash/maximum capacity for the trash, and if it is full (or if the instance is too large) then moving to the trash might fail, even if trash is present and enabled.

Reply:
> Maybe, but there are other fairly common reasons depending on the OS. For example, on Windows I believe there is a maximum file size for items in the trash/maximum capacity for the trash, and if it is full (or if the instance is too large) then moving to the trash might fail, even if trash is present and enabled.

hmm, that's why maybe I could try using a temporary file to test the Trash

Reply:
> but that wouldn't be very elegant

True, and it would fail in the case where the failure reason is that the instance is too big (a temporary file is nowhere near as large!).

Usually, I've seen file managers handle it like this: if the "deletion" can be done by moving to the trash (or can otherwise be undone), then it would occur immediately without a confirmation. A confirmation will appear, however, if the deletion will be permanent for any reason.

If you go that route, there should probably also be some clear way to undo the deletion from PolyMC itself without having the user manually go into the trash. (Actually, I like this idea regardless, if it's not too difficult to implement. If I delete some file/folder by accident, I usually just press the undo shortcut on my keyboard and that usually recovers it from the trash – being able to do the same thing in PolyMC would be nice.)

Reply:
That's a good idea. Although implementing Cmd-Z will take some time 😬

Reply:
@kthchew part 1 is done

Reply:
Voila

[`02f30de` (#549)](https://github.com/PolyMC/PolyMC/pull/549/commits/02f30de1c11fa53e61c199032a3280afe618e3b2)